### PR TITLE
Replace Boost.Test with Catch2

### PR DESCRIPTION
**Pull request by ax3l**
Replace [Boost.Test](https://www.boost.org/doc/libs/1_66_0/libs/test/doc/html/index.html) with [Catch2](https://github.com/catchorg/Catch2).

Step toward #70 and #86

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -563,6 +579,7 @@ message("")
 message("  Additionally installed third party libraries:")
 invertBoolean(openPMD_USE_INTERNAL_VARIANT openPMD_HAVE_EXTERNAL_VARIANT)
 message("    MPark.Variant: ${openPMD_HAVE_EXTERNAL_VARIANT}")
+message("           Catch2: OFF")
```
Comment (anokfireball) [inline CMakeLists.txt:588]: Should this do the same inversion as the lines above with ```openPMD_USE_INTERNAL_CATCH```?

**Review by anokfireball — APPROVED** [review]
Currently only a change in dependencies, but Catch2 will allow for more readable and maintainable tests.

File: CMakeLists.txt (hunk 1 above)
Comment (ax3l) [reply CMakeLists.txt:588]: Good question! No, Catch2 will not be installed on `make install` simply because it is only needed at build-time.

Comment (anokfireball) [reply CMakeLists.txt:588]: It took me a while to get the intent of the messages.
That is a really elegant and clever solution and makes the Variant output really useful.

Comment (ax3l) [reply CMakeLists.txt:588]: Haha, thanks.

**Pull request merged**